src/common/libflux/message.c

```diff
@@ -359,35 +527,40 @@ flux_msg_t *flux_msg_decode (const void *buf, size_t size)
         }
         if (!(zf = zframe_new (p, n)))
             goto nomem;
-        if (zmsg_append (msg->zmsg, &zf) < 0)
+        if (zmsg_append (zmsg, &zf) < 0)
             goto nomem;
         p += n;
     }
+    if (zmsg_to_msg (msg, zmsg) < 0)
```

Comment:
Maybe this is an intentional baby step, but there seems to be a disparity between `flux_msg_encode()` which is zmsg/zframe free, and `flux_msg_decode()` which is not.  Is there a good reason not to go all the way and convert the decode path as well?

Reply:
I actually have a branch that did remove `zmsg` use from `flux_msg_decode()`, using a `zlist` instead (b/c that's all I'm really using `zmsg` for in `flux_msg_decode()`, just as a list) and I created a `zlist_to_msg()` kinda function.

But `zmsg_to_msg()` is still needed for `flux_msg_recvzsock()`.  And a (good) common function between `zlist_to_msg()` and `zmsg_to_msg()` was annoying enough I just abandoned the `zlist_to_msg()` until the longer term plan was established.

We could go ahead and add `zlist_to_msg()` here b/c I think we're going to end up having two functions in the end anyways (one in `libflux` and one in `librouter`).

Reply:
In the interest of keeping things moving forward, maybe we should leave as is.  This is a big change and there will be opportunities to refine it later.